I`m not sure if it is necessary to repeat previous commit message

MaxymRomanchuk opened the pull request:
Anyway, stars have random coordinates and size (so the speed) when appear.
Stars "shrink" and dissappear before sunrise.
Each star has its target position in "Botany" sign and needs some time to get to it.

MaxymRomanchuk committed:
I`m not sure if it is necessary to repeat previous commit message
Anyway, stars have random coordinates and size (so the speed) when appear.
Stars "shrink" and dissappear before sunrise.
Each star has its target position in "Botany" sign and needs some time to get to it.

LudwigVonChesterfield commented on source/changed/Stars.pde:
```diff
@@ -0,0 +1,72 @@
+    this.SpeedY = this.AccY*dt;
+    
+    if(this.pos.x*this.kX<this.target.x*this.kX) {
+      this.pos.x += this.kX*this.SpeedX;
```
🤔 Мне кажется kX надо было бы умножать на ускорение, в функции с ускорениями, и тогда бы не пришлось переписывать applySpeed.

LudwigVonChesterfield commented on source/changed/Stars.pde:
```diff
@@ -0,0 +1,72 @@
+  public void setColor() 
+  {
+    this.visibleSize = map(sun.Light,0,40*(this.Size/3),this.Size,1);
+    this.Color = color(0,0,map(sun.Light,100.0,0.0,178.5,255.0));
```
думаю если у звёздочек будет самую малость разные цвета(но не насыщенные) было бы классно.

LudwigVonChesterfield commented on source/changed/source.pde:
```diff
@@ -0,0 +1,203 @@
+  for(int i = 0;i<BotanySign.length;i++)
+  {
+    new Stars(new PVector(BotanySign[i][0],BotanySign[i][1]),new PVector(random(width),random(120,height/2)));
+    println(i);
```
вот этого в продакшене быть не должно

LudwigVonChesterfield commented on source/changed/source.pde:
```diff
@@ -0,0 +1,203 @@
+  */
+
+  background(150.0, min(sun.Light + 25.0, 230.0), min(sun.Light + 25.0, 230.0));
+  if(mousePressed) 
```
как и вот этого

LudwigVonChesterfield reviewed (changes requested):
И вообще, почему-то файлы находятся в подпапке /changed/, не знаю зачем.

EDIT: если не будет решено до вторника - доделаю сам.

LudwigVonChesterfield commented on source/changed/Stars.pde:
```diff
@@ -0,0 +1,72 @@
+  
+  public Stars(PVector dest, PVector start) {
+    super(start);
+    startPos = start.copy();
```
what's the point of copying a vector you initiate just for the star.

LudwigVonChesterfield commented on source/changed/Stars.pde:
```diff
@@ -0,0 +1,72 @@
+  private float Size;
+  private float visibleSize = 2;
+  
+  public Stars (PVector start) {
```
the space

(aaaaaaaaaaaaaaaaaaaaaaaaah)

MaxymRomanchuk committed:
Found out that ArrayList has better perfomance than even array,
so I made one for stars to draw them "behind" all other objects.
Added few new stars.

LudwigVonChesterfield commented on source/source.pde:
```diff
@@ -76,7 +91,15 @@ void draw()
   hslSetColor(background_hue, background_satur, background_light);
   */
-
+  if(mousePressed) {
```
Debug messages should never get to prod.

LudwigVonChesterfield commented on source/Stars.pde:
```diff
@@ -0,0 +1,64 @@
+  
+  public Stars(PVector dest, PVector start) {
+    super(start);
+    startPos = start.copy();
```
Considering that you're initiating a new Vector for each Star spawned, copying the object isn't desirable.

LudwigVonChesterfield committed:
Adding spaces, Star Acceleration-Speed relation correction.

Star shimmering, different star colors.
